alacritty_terminal/src/term/mod.rs

```diff
@@ -281,6 +281,52 @@ impl<'a, C> RenderableCellsIter<'a, C> {
             cursor_style,
         }
     }
+
+    /// Check selection state of a cell.
+    fn is_selected(&self, point: Point) -> bool {
+        let selection = match self.selection {
+            Some(selection) => selection,
+            None => return false,
+        };
+
+        // Point itself is selected
+        if selection.contains(point.col, point.line) {
+            return true;
+        }
+
+        let num_cols = self.grid.num_cols().0;
+        let cell = self.grid[&point];
+
+        // Check if wide char's spacers are selected
+        if cell.flags.contains(Flags::WIDE_CHAR) {
+            let prevprev = point.sub(num_cols, 2, false);
+            let prev = point.sub(num_cols, 1, false);
+            let next = point.add(num_cols, 1, false);
+
+            // Check trailing spacer
+            return selection.contains(next.col, next.line)
+                // Check line-wrapping, leading spacer
+                || (self.grid[&prev].flags.contains(Flags::WIDE_CHAR_SPACER)
+                    && !self.grid[&prevprev].flags.contains(Flags::WIDE_CHAR)
+                    && selection.contains(prev.col, prev.line));
+        }
+
+        // Check if spacer's wide char is selected
+        if cell.flags.contains(Flags::WIDE_CHAR_SPACER) {
+            let prev = point.sub(num_cols, 1, false);
+
+            if self.grid[&prev].flags.contains(Flags::WIDE_CHAR) {
+                // Check previous cell for trailing spacer
+                return self.is_selected(prev);
+            } else {
+                // Check next cell for line-wrapping, leading spacer
+                let next = point.add(num_cols, 1, false);
+                return self.is_selected(next);
+            }
+        }
+
+        false
+    }
```

Comment:
I think the current approach should be pretty good for perf actually, because in most cases we exit early and never go inside of the WIDE_CHAR and WIDE_CHAR_SPACER blocks.

So we just have two conditionals that are absolutely necessary anyways and everything else is skipped. Except for the `self.grid[&point]` and `num_cols` parts, but maybe the optimizer even helps with that, I don't know. They shouldn't impact performance much though.

But the main goal is making things more readable, I think that's definitely worth it here. The performance impact should not be measurable, I certainly can't.